File: BOJ_Practice/1406.cpp
// same to pyfile
// time exceeded

#include <iostream>
#include <cstdio>
#include <string>
using namespace std;

int main(int argc, char* argv[]) {

	string data;
	cin >> data;
	
	int cnt;
	cin >> cnt;
	int cur = data.length();

	string cmd;
	cin.ignore(INT_MAX, '\n');
	for (size_t i = 0; i < cnt; i++) {
		getline(cin, cmd);
		if (cmd[0] == 'L') {
			if (cur > 0) cur--;
		}
		else if (cmd[0] == 'D') {
			if (cur < data.length())
				cur++;
		}
		else if (cmd[0] == 'B') {
			if (cur > 0) {
				cur--;
				data.erase(data.begin() + cur);
			}
		}
		else if (cmd[0] == 'P') {	// P
			data.insert(cur, cmd.substr(2));
			cur++;
		}
	}
	cout << data;


	return 0;
}

File: BOJ_Practice/1406.py
a= list(input())
cnt= int(input())
cur= len(a)
# abcd_ cur=4
# _ cur=0
# abc(d) cur= 3 >> abce(d) cur=4

for i in range(0, cnt):
    cmd= input()
    cmd= cmd.split(' ')
    if cmd[0] == 'L':
        if cur == 0: pass
        else: cur-=1
    elif cmd[0] == 'D':
        if cur == len(a): pass
        else: cur+=1
    elif cmd[0] == 'B':
        if cur == 0: pass
        else:
            cur-=1
            del(a[cur])
    elif cmd[0] == 'P':
        a.insert(cur, cmd[1])
        cur+=1
s=''
for i in a:
    s+=i
print(s)